contract/pkg/client.go

```diff
@@ -67,6 +72,57 @@ func CreateBlockchainClient(apiUrl string) BlockchainClient {
 	}
 }
 
+func (b *blockchainClient) ListenContractEvents(contractAddressSS58 string, f ContractEventHandler) error {
+	meta, err := b.RPC.State.GetMetadataLatest()
+	if err != nil {
+		return err
+	}
+
+	contract, err := DecodeAccountIDFromSS58(contractAddressSS58)
+	if err != nil {
+		return err
+	}
+
+	key, err := types.CreateStorageKey(meta, "System", "Events", nil, nil)
+	if err != nil {
+		return err
+	}
+
+	sub, err := b.RPC.State.SubscribeStorageRaw([]types.StorageKey{key})
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		defer sub.Unsubscribe()
+		for {
+			evt := <-sub.Chan()
```

Comment:
I remember each system event has `uint64` index for detecting missed events. Did you check this, so we can be sure that didn't miss any event from blockchain?

Reply:
I didn't find this index anywhere. In my local tests, the events subscription is quite stable, but anyway, missing an event is not a problem - it will fallback to the previous (cache expiration) behaviour.